kubespawner/objects.py

```diff
@@ -829,31 +831,43 @@ def make_ingress(
         labels=ingress_labels,
     )
 
+    ingress_specifications = ingress_specifications or [{}]
+
+    rules = [
+        V1IngressRule(
+            host=spec.get("host"),
+            http=V1HTTPIngressRuleValue(
+                paths=[
+                    V1HTTPIngressPath(
+                        path=path,
+                        path_type="Prefix",
+                        backend=V1IngressBackend(
+                            service=V1IngressServiceBackend(
+                                name=name,
+                                port=V1ServiceBackendPort(
+                                    number=target_port,
+                                ),
+                            ),
+                        ),
+                    ),
+                ],
+            ),
+        )
+        for spec in ingress_specifications
+    ]
+
+    tls = [
+        V1IngressTLS(hosts=[spec["host"]], secret_name=spec["tlsSecret"])
+        for spec in ingress_specifications
+        if "tlsSecret" in spec
+    ]
+
     ingress = V1Ingress(
         kind='Ingress',
         metadata=ingress_meta,
         spec=V1IngressSpec(
-            rules=[
-                V1IngressRule(
-                    host=host,
```

Comment:
It seems that `host` is a variable that we figure out based on `routespec`, but then later won't be using. I figure it should be cleaned up or similar if not needed.

Also related, now you are required to provide ingress_specifications, right? Otherwise you will attempt to create the V1IngressSpec with rules=None and tls=None - does that make sense to do?

If you are required to provide something, it would be good to error more verbosly if you aren't provided with that.

Reply:
Ok, I've set `host` variable as a primary source of hostname, and `spec["host"]` as a fallback

Reply:
`ingress_specifications` is not required:
```python
ingress_specifications = ingress_specifications or [{}]
```
If `ingress_specifications` is None or empty list, `[... for spec in ingress_specifications]` will return a list with one item